File: hrms-frontend/src/services/announcementService.js
// src/services/announcementService.js
import api from '../api/axiosConfig';

const announcementService = {
  getAll: () => api.get('/announcements'),
  getForEmployee: (empId) => api.get(`/announcements/employee/${empId}`),
  getById: (id, empId) =>
    api.get(`/announcements/${id}${empId ? `?employeeId=${empId}` : ''}`),
  create: (data) => api.post('/announcements', data),
  update: (id, data) => api.put(`/announcements/${id}`, data),
  delete: (id) => api.delete(`/announcements/${id}`),
  markAsRead: (id, empId) =>
    api.post(`/announcements/${id}/read/${empId}`),
  togglePin: (id) => api.patch(`/announcements/${id}/toggle-pin`),
  toggleActive: (id) => api.patch(`/announcements/${id}/toggle-active`),
};

export default announcementService;

File: hrms-backend/src/main/java/com/hrms/service/AnnouncementService.java

//service/AnnouncementService.java
package com.hrms.service;

import com.hrms.dto.request.AnnouncementRequest;
import com.hrms.dto.response.AnnouncementResponse;

import java.util.List;

public interface AnnouncementService {
 AnnouncementResponse create(AnnouncementRequest request, Long userId);
 AnnouncementResponse update(Long id, AnnouncementRequest request);
 AnnouncementResponse getById(Long id, Long employeeId);
 List<AnnouncementResponse> getAll();
 List<AnnouncementResponse> getForEmployee(Long employeeId);
 void delete(Long id);
 void markAsRead(Long announcementId, Long employeeId);
 void togglePin(Long id);
 void toggleActive(Long id);
}